### DISCUSSION TITLE
Model inference performance slowdown in iOS 16

### DISCUSSION THREAD

**Discussion by titanium-cranium** [ultralytics/yolov5#9850] [Q&A]
Hi All:

I am building a react-native app for iPhone that uses a YOLOv5s model exported to coreml for labelling video frames on the fly as the camera is recording.  Recently I upgraded my phone to iOS16 and discovered that the benchmark around the model inference component of the native Swift code is running substantially slower.  Prior to upgrading, inference with iOS 15.6 was ~14ms, with iOS16 it's more like 60ms. 

Has anyone else encountered a similar issue?  Any ideas on what can be done? 

Thanks,

Brendan

**Comment by glenn-jocher** [comment]
YOLOv5 CoreML model inference speeds are unchanged in iOS 16. You can view example speeds with the iDetection app: https://apps.apple.com/us/app/idetection/id1452689527

@sergiossm let's get the inference speeds and FPS on the Ultralytics app similar to how it's shown in iDetection so we can start pointing users there :)

**Reply by titanium-cranium** [reply]
Turned out something about 16.0.1-3 caused the issue for us.  Upgrading to 16.1.0 restored it back to full speed.  14ms inference on a full HD image.  ⚡

**Reply by glenn-jocher** [reply]
Good to know!
